core/vm/evm.go

```diff
@@ -177,6 +178,13 @@ func (evm *EVM) Interpreter() *EVMInterpreter {
 // the necessary steps to create accounts and reverses the state in case of an
 // execution error or failed value transfer.
 func (evm *EVM) Call(caller ContractRef, addr common.Address, input []byte, gas uint64, value *big.Int) (ret []byte, leftOverGas uint64, err error) {
+	// Capture the tracer start/end events in debug mode
+	if evm.Config.Tracer != nil {
+		evm.captureBegin(evm.depth == 0, CALL, caller.Address(), addr, input, gas, value)
+		defer func(startGas uint64) {
+			evm.captureEnd(evm.depth == 0, CALL, startGas, leftOverGas, ret, err)
+		}(gas)
+	}
```

Comment:
You've moved this up: this will be changed behaviour regarding e.g. when trying to send more balance than one has. Previously, it would error early and not invoke the tracer, now you invoke it. 

Is that intentional?

Reply:
Ping?

Reply:
Sorry. Yes it is intentional. We did it for all variants of CALL and CREATE. This is not necessarily related to the live tracing feature but it came up there are some errors the tracers can't capture.

Reply:
It's not so much that the tracers can't capture the error, in my opinion. It is more semantics. Let's consider the case of doing a call with 10 ether when you only have 1. 

With this change, it is considered that a call was made: started and subsequently ended with an error due to insufficient balance. 

Before this change: it is considered that a call was never made; it is not possible to send 10 ether when you only have 1.  


I'm not convinced that the new semantics are better... ?

Reply:
E.g. see this issue: https://github.com/ethereum/go-ethereum/issues/25953#issuecomment-1303118721

I think the new semantics are better. Let's say there's an CALL opcode. It can be strange for users to see no call trace after this. Also there is no way to know what happened, why the call didn't start.

Reply:
After considering it a bit more, yeah I agree

Reply:
Ok so now `OnNewAccount` event is removed. As per the discussion above, whether an account exists or not is only relevant to the trie structure. What consumers care about is balance, nonce, code and storage of an account. I believe all use-cases of this method can be achieved in other ways.